blog.md

```diff
@@ -0,0 +1,258 @@
+  root: PgId,
+  /// monotonically incrementing, used by NextSequence()
+  sequence: u64,
+}
+
+/// `Meta` represents the on-file layout of a database's metadata
+/// 
+/// `meta` in Go BBolt
+#[repr(C)]
+pub struct Meta {
+  /// Uniquely ID for BBolt databases 
+  magic: u32,
+  /// Database version number
+  version: u32,
+  /// Database page size where page address = [PgId] * meta.page_size
+  page_size: u32,
+  flags: u32,
+  /// Root bucket header
+  root: BucketHeader,
+  /// FreeList page location
+  free_list: PgId,
+  /// The end of the database where EOF = meta.pgid * meta.page_size
+  pgid: PgId,
+  /// Current transaction ID
+  txid: TxId,
+  /// Checksum of the previous Meta fields using the 64-bit version of the Fowler-Noll-Vo hash function
+  checksum: u64,
+}
+
+/// `LeafPageElement` represents the on-file layout of a leaf page's element
+/// 
+/// `leafPageElement` in Go BBolt 
+#[repr(C)]
+pub struct LeafPageElement {
+  /// Additional flag for each element. If leaf is a Bucket then 0x01 set
+  flags: u32,
+  /// The distance from this element's pointer to its key/value location
+  pos: u32,
+  /// Key length
+  key_size: u32,
+  /// Value length
+  value_size: u32,
+}
+
+///`BranchPageElement` represents the on-file layout of a branch page's element
+/// 
+/// `branchPageElement` in Go BBolt
+#[repr(C)]
+pub struct BranchPageElement {
+  /// The distance from this element's pointer to its key location
+  pos: u32,
+  /// Key length
+  key_size: u32,
+  /// Page ID of this branch's
+  pgid: PgId,
+}
+
+```
+
+The 4 database page types are also straightforward. 
+
+Page Layouts
+```
+      Meta Page
+      ┌──────┐
+      │ Page │
+      │Header│
+      ├──────┤
+      │ Meta │
+      └──────┘
+
+
+   Freelist Page
+      ┌──────┐
+      │ Page │
+      │Header│
+      ├──────┤     First element will be the FreeList
+      │PgId 0│◀────   count if PageHeader.count ==
+      ├──────┤                  U16::MAX
+      │PgId 1│
+      ├──────┤
+      │PgId 2│
+      ├──────┤
+      │ .... │
+      └──────┘
+
+ Leaf Element Page
+      ┌──────┐
+      │ Page │
+      │Header│
+      ├──────┤
+      │Elem 0│
+      ├──────┤
+      │Elem 1│       key ptr =
+      ├──────┤  elem ptr + elem.pos
+      │Elem 2│   ┌─── ┌───┐
+      ├──────┤   │    │ k │
+      │ .... │   │    │ e │
+      ├──────┤───┘    │ y │
+      │K/V 0 │        │ v │
+      ├──────┤───┐    │ a │
+      │K/V 1 │   │    │ l │
+      ├──────┤   │    │ u │
+      │K/V 2 │   │    │ e │
+      ├──────┤   └─── └───┘
+      │ .... │
+      └──────┘
+
+Branch Element Page
+      ┌──────┐
+      │ Page │
+      │Header│
+      ├──────┤
+      │Elem 0│
+      ├──────┤
+      │Elem 1│
+      ├──────┤
+      │Elem 2│
+      ├──────┤
+      │ .... │
+      ├──────┤           key ptr =
+      │Key 0 │◀──── elem ptr + elem.pos
+      ├──────┤
+      │Key 1 │
+      ├──────┤
+      │Key 2 │
+      ├──────┤
+      │ .... │
+      └──────┘
+
+Created with Monodraw
```

Comment:
Good attribution, but pull it out of the codeblocks and/or provide the info in footer. It's essentially tertiary information.